conductor/submitter.py

```diff
@@ -1768,92 +1665,636 @@ class SubmitterPrefs(pyside_utils.UiFilePrefs):
+        gpu_model, gpu_count = _gpu["gpu_model"], _gpu["gpu_count"]
+        cores, memory = instance['cores'], instance['memory']
+        cores_memory = "%s/%s" % (cores, memory)
+
+        # enable/disable the gpu checkbox (based upon whether the instance type has a gpu
+        self.ui_gpu_chkbx.setChecked(bool(gpu_model))
+
+        # If the instance type has a gpu, then populate the ui comboboxes to reflect the gpu info
+        if gpu_model:
+            self.setGpuModel(gpu_model)
+            self.populateGpuCountCmbx(self._combobox_map[gpu_model])
+            self.setGpuCount(gpu_count)
+
+        # Populate the cores/memory combobox
+        self.populateCoresMemoryCmbx(self._combobox_map[gpu_model][gpu_count])
+
+        # Set the active cores/memory entry to reflect the same value as the instance type.
+        self.setCoresMemory(cores_memory)
+
+    def _gpuCheckboxToggled(self, checked):
+        '''
+        When the gpu checkbox changes...
+          1. Enable/disable the gpu widget (reflect the gpu checkbox)
+          2. Populate the cores/memory combobox (reflect whether gpus should be included or not)
+          4. artificial trigger cores/memory change (use current index)
+        '''
+        self.ui_gpu_type_wgt.setEnabled(checked)
+        # If the gpu checkbox is checked on then populate the memory/cores combobox with gpu instances.
+        if checked:
+            mapping = self.ui_gpu_count_cmbx.currentData(role=self.ROLE_CHILD_DATA)
+            self.populateCoresMemoryCmbx(mapping)
+        # otherise populate the memory/cores combobox with non-gpu instances
+        else:
+            self.populateCoresMemoryCmbx(self._combobox_map[None][0])
+
+        # trigger the the cores/memory combobox to trigger/update the rest of ui.
+        self.ui_cores_memory_cmbx.currentIndexChanged.emit(self.ui_cores_memory_cmbx.currentIndex())
+
+    # -----------------------------------
+    # GPU MODEL
+    # -----------------------------------
+
+    def setGpuModel(self, gpu_model, block_signals=True):
+        '''
+        Set the gpu model combobox's active item to that of the the given `gpu_model` (the KEY value).
+        '''
+        self._setActiveComboboxItem(
+            self.ui_gpu_model_cmbx,
+            gpu_model,
+            self.ROLE_KEY,
+            block_signals=block_signals,
+        )
+
+    def populateGpuModelCmbx(self, data, block_signals=True):
+        '''
+        Populate the Gpu Model combobox with the given data.
+        '''
+        # Populate Gpus dropdown
+        items = []
+        for gpu_model, gpu_count_mapping in data.iteritems():
+            if gpu_model:
+                gpu = self._gpu_map[gpu_model]
+                tooltip = "{gpu_architecture} {gpu_model} GPU with {gpu_cuda_cores} Cuda cores and {gpu_memory}GB memory"
+                items.append({
+                    # TODO(lws): Remove hard-coded "nvidia". The vendor name should be provided via backend
```

Comment:
@flebel I'm not sure what that link is supposed to tell me?  I'm aware of the `gpu_architecture` field (I use it 3 lines above).  The problem I have with it is that it includes the vendor name (`"Nvidia Turing`", rather than just `"Turing"`)
![image](https://user-images.githubusercontent.com/10409070/102327508-b1e28e80-3f3a-11eb-81cc-6cf2e19917ec.png)

I would suggest that the instance-types endpoint return a gpu data structure as defined in ae: https://github.com/AtomicConductor/conductor_ae/blob/master/app_engine/conductor_app/app_constants/pricing.py#L11-L68, e.g. 
```python
{
        "arch": "Turing",
        "cores_cuda": 2560,
        "cores_tensor": 320,
        "launch": 2018,
        "memory": 16,
        "model": "T4",
        "vendor": "Nvidia",
}
```

Note that

- the `model` value doesn't include the `"Tensor"` string  (I don't know why it ever did).
- there is a discreet `vendor` field

A structure like this gives us the most flexibility when designing ux for customer interaction.

Reply:
Yep, you can make the change in the cloud resources, and include the vendor name in a new description field that's meant for the text displayed to the end users, e.g. `NVIDIA T4 16GB`.